lsstseries/timeseries.py

```diff
@@ -94,3 +96,141 @@ def _build_index(self, band):
         tuples = zip(band, range(len(band)))
         index = pd.MultiIndex.from_tuples(tuples, names=["band", "index"])
         return index
+
+    def stetson_J(self, band=None):
+        """Compute the stetsonJ statistic on data from one or several bands
+
+        Parameters
+        ----------
+        band : `str` or `list` of `str`
+            Single band descriptor, or list of such descriptors.
+
+        Returns
+        -------
+        stetsonJ : `dict`
+            StetsonJ statistic for each of input bands.
+
+        Notes
+        ----------
+        In case that no value for band is passed, the function is executed
+        on all avaliable bands.
+        """
+
+        if band is None:
+            unq_band = np.unique(self.band)
+        if type(band) == 'str':
```

Comment:
This should be str, not 'str', doesn't work as intended. Also, should handle case where user provides a subset of bands (e.g. band=['g','z']), as currently that would never assign unq_band and it would error out

Reply:
I have modified the code so that it supports passing a subset of bands - including the possibility to pass bands that do not exist in the data. I have incorporate that with if..else.. statment; not sure if this is long-term feasible.

> This looks good overall, two things:
> 
> 1. Is it worth putting special handling for the case where you have 0 or 1 data points? In particular, should they return a NaN? I tried it with a dummy example and ran into that for the N=1 case, because it attempts an N/N-1 calculation. Just thinking about the use case where we have O(10^6+) running and don't want these to be erroring out necessarily
> 2. stetson_j_single has required parameters of fluxes and errors, meaning that a user would need to filter down to those band specific values manually for ingest into the function, would a single argument for band be better? This aligns it with stetson_J, but also would potentially make it redundant as it looks like stetson_J can handle a single input band string